pkg/netconfig/netconfig.go

```diff
@@ -102,21 +105,45 @@ func IPTablesLoggingChainRule(ipVersion string, protocol string, ipSet string, d
 
 func AddIPTablesLoggingRules(ipVersion, ipSet, defaultNetworkDevice string, blockIngress bool) error {
 
-	if err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, true, blockIngress); err != nil {
-		err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, false, blockIngress)
+	if err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, true, false, blockIngress); err != nil {
+		err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, false, false, blockIngress)
 		if err != nil {
 			return fmt.Errorf("error creating tcp logging chain rules for %s, device %s %v", ipSet, defaultNetworkDevice, err)
 		}
 	}
-	if err := IPTablesLoggingChainRule(ipVersion, "udp", ipSet, defaultNetworkDevice, true, blockIngress); err != nil {
-		err := IPTablesLoggingChainRule(ipVersion, "udp", ipSet, defaultNetworkDevice, false, blockIngress)
+	if err := IPTablesLoggingChainRule(ipVersion, "udp", ipSet, defaultNetworkDevice, true, false, blockIngress); err != nil {
+		err := IPTablesLoggingChainRule(ipVersion, "udp", ipSet, defaultNetworkDevice, false, false, blockIngress)
 		if err != nil {
 			return fmt.Errorf("error creating udp logging chain rules for %s, device %s %v", ipSet, defaultNetworkDevice, err)
 		}
 	}
 	return nil
 }
 
+func RemoveIPTablesLoggingRules(ipVersion, ipSet, defaultNetworkDevice string) error {
+	// we don't care if SYN filtering was enabled previously. delete both variants.
+	if err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, true, false, true); err == nil {
+		err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, false, true, true)
+		if err != nil {
+			return fmt.Errorf("error deleting tcp logging chain rules for %s, device %s %v", ipSet, defaultNetworkDevice, err)
+		}
+	}
+	if err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, true, false, false); err == nil {
+		err := IPTablesLoggingChainRule(ipVersion, "tcp", ipSet, defaultNetworkDevice, false, true, false)
+		if err != nil {
+			return fmt.Errorf("error deleting tcp logging chain rules for %s, device %s %v", ipSet, defaultNetworkDevice, err)
+		}
+	}
```

Comment:
The `if` blocks are almost the same. This is somewhat confusing. Please add at least some comment and adapt the error message, which should not be the same.

Reply:
there is a comment in line 124. You have a point about the error message, I will change it.